docs/tutorials/shors-algorithm.ipynb

```diff
@@ -0,0 +1,1334 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "59ae21fc-e3d0-48e3-8cc6-dbce59c747ca",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "            Register Output           Phase\n",
+      "0  00000000(bin) =   0(dec)    0/256 = 0.00\n",
+      "1  01000000(bin) =  64(dec)   64/256 = 0.25\n",
+      "2  10000000(bin) = 128(dec)  128/256 = 0.50\n",
+      "3  11000000(bin) = 192(dec)  192/256 = 0.75\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Rows to be displayed in table\n",
+    "rows = []\n",
+    "# Corresponding phase of each bitstring\n",
+    "measured_phases = []\n",
+    "\n",
+    "for output in counts:\n",
+    "    decimal = int(output, 2)  # Convert bitstring to decimal\n",
+    "    phase = decimal / (2**num_control)  # Find corresponding eigenvalue\n",
+    "    measured_phases.append(phase)\n",
+    "    # Add these values to the rows in our table:\n",
+    "    rows.append(\n",
+    "        [\n",
+    "            f\"{output}(bin) = {decimal:>3}(dec)\",\n",
+    "            f\"{decimal}/{2 ** num_control} = {phase:.2f}\",\n",
+    "        ]\n",
+    "    )\n",
+    "\n",
+    "# Print the rows in a table\n",
+    "headers = [\"Register Output\", \"Phase\"]\n",
+    "df = pd.DataFrame(rows, columns=headers)\n",
+    "print(df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f0ca65bb-abb4-4f4b-bc32-4749f36e5780",
+   "metadata": {},
+   "source": [
+    "Recall that the any measured phase corresponds to $\\theta = k / r$ where $k$ is sampled uniformly at random from $\\{0, 1, \\dots, r-1 \\}$. Therefore, we can use the continued fractions algorithm to attempt to find $k$ and the order $r$. Python has this functionality built in. We can use the `fractions` module to turn a float into a `Fraction` object, for example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "999f04bc-f912-465f-9cba-90d08bda3758",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Fraction(5998794703657501, 9007199254740992)"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "Fraction(0.666)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5452b79-967f-4fe9-86d7-f9797987a8b5",
+   "metadata": {},
+   "source": [
+    "Because this gives fractions that return the result exactly (in this case, `0.6660000...`), this can give gnarly results like the one above. We can use the `.limit_denominator()` method to get the fraction that most closely resembles our float, with denominator below a certain value:"
```

Comment:
```suggestion
    "Because this gives fractions that return the result exactly (in this case, `0.6660000...`), this can give gnarly results like the one above. We can use the `.limit_denominator()` method to get the fraction that most closely resembles our float, with a denominator below a certain value:"
```